src/plugins/discover/public/application/main/components/sidebar/discover_field_visualize.tsx

```diff
@@ -37,9 +37,12 @@ export const DiscoverFieldVisualize: React.FC<Props> = React.memo(
       return null;
     }
 
-    const handleVisualizeLinkClick = (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
+    const handleVisualizeLinkClick = async (
+      event: React.MouseEvent<HTMLAnchorElement, MouseEvent>
+    ) => {
       // regular link click. let the uiActions code handle the navigation and show popup if needed
       event.preventDefault();
+
       trackUiMetric?.(METRIC_TYPE.CLICK, 'visualize_link_click');
       triggerVisualizeActions(visualizeInfo.field, dataView.id, details.columns);
```

Comment:
I think we need to ask the user to persist the ad-hoc for geo coordinates, some info how to detect which trigger is used:

https://github.com/elastic/kibana/blob/bcbef78b4dfaebe40f334fcaa5cef94fda9acc29/src/plugins/discover/public/application/main/components/sidebar/lib/visualize_trigger_utils.ts#L19-L24

because currently the maps is displayed, and it works, but e.g. persistence doesn't work 

also, until resolved, the navigation to Lens doesn't work:
<img width="988" alt="Bildschirmfoto 2022-08-31 um 20 59 26" src="https://user-images.githubusercontent.com/463851/187763497-b674e4c2-bf28-4ce9-8889-812b70c30524.png">
The sidebar breaks, because the currentIndexPattern is undefined when going to Lens with an ad-hoc one 

https://github.com/elastic/kibana/blob/1c191166b3ea96e83ed931285b321f26bb1a56b3/x-pack/plugins/lens/public/indexpattern_datasource/datapanel.tsx#L309

FYI dear @elastic/kibana-vis-editors , guess the `indexPatterns` are is considering the ad-hoc one to be a stranger

Reply:
I think the data view cache is cleared when navigating - we should pass the spec in the action instead of the id. Of course that will require changes on the Lens side as well, happy to split that out into a separate PR.

Reply:
Visualise navigation fixed